manifests/init.pp

```diff
@@ -122,48 +122,29 @@
 # @author https://github.com/elastic/puppet-logstash/graphs/contributors
 #
 class logstash(
-  $ensure            = 'present',
-  $status            = 'enabled',
-  $restart_on_change = true,
-  $auto_upgrade       = false,
-  $version           = undef,
-  $package_url       = undef,
-  $package_name      = 'logstash',
-  $download_timeout  = 600,
-  $logstash_user     = 'logstash',
-  $logstash_group    = 'logstash',
-  $config_dir         = '/etc/logstash',
-  $purge_config      = true,
-  $service_provider  = undef,
-  $settings          = {},
-  $startup_options   = {},
-  $jvm_options       = [],
-  $manage_repo       = true,
-  $repo_version      = '5.x',
-)
+  Enum['present','absent'] $ensure                         = 'present',
```

Comment:
I really want to deprecate Puppet 3 syntax too, but I fear there are a lot of sites out there that (still) haven't made the jump.

When we drop Puppet 3 support, we'll want to do it in a distinct and easily communicated way. A way, that in particular, won't surprise people.

We're currently entertaining the idea of dropping Puppet 3 support with the release of Elastic Stack (and module version) 6.